lib/config.py

```diff
@@ -149,46 +49,61 @@ def __init__(self):
+        self._setup_command_line_parser(CONFIG_METADATA_FILE_PATH)
 
     @property
     def CONF(self):
         return self._CONF
 
 
-    def _setup_command_line_parser(self, subcommands):
+    def _setup_command_line_parser(self, config_metadata_file_path):
         """
         Configures the argument parser object to match the expected
             configuration.
 
         Args:
-            subcommands ([(str, str, [dict])]): subcommands. Each subcommand is
-                a tuple with subcommand name, subcommand help message, subcommand arguments.
-                Each subcommand argument is a dict from a tuple with argument possible names
-                to a dict with argument attributes. See ArgParse documentation for possible
-                values in this dict.
+            config_metadata_file_path (str): config metadata file path
         """
         self.parser.add_argument('--config-file', '-c',
                                  help='Path of the configuration file for build '
-                                      'scripts')
-        self.parser.add_argument('--verbose', '-v',
-                                 help='Set the scripts to be verbose',
-                                 action='store_true')
-        self.parser.add_argument('--log-size',
-                                 help='Size in bytes above which the log file '
-                                 'should rotate', type=int)
-        self.parser.add_argument('--work-dir', '-w',
-                                 help='Directory used to store all temporary '
-                                 'files created during the process.')
+                                      'scripts',
+                                 # NOTE(maurosr): move this to /etc in the future
+                                 default='./config.yaml')
 
         subparsers = self.parser.add_subparsers(
             dest="subcommand",
             help="Available subcommands")
-        for command, help_msg, arg_groups in subcommands:
-            command_parser = subparsers.add_parser(command, help=help_msg)
-            for arg_group in arg_groups:
-                for arg, options in arg_group.items():
-                    command_parser.add_argument(*arg, **options)
+
+        # setup parser from config metadata file
+        with open(config_metadata_file_path) as fh:
+            config_metadata = yaml.safe_load(fh)
+        for section in config_metadata:
+            if section != "common":
+                command = section.replace("_", "-")
+                command_help = config_metadata[section]["help"]
+                command_parser = subparsers.add_parser(command, help=command_help)
+                target_parser = command_parser
+            else:
+                target_parser = self.parser
+            config_metadata[section].pop("help")
+            for attribute in config_metadata[section]:
+                argument = attribute.replace("_", "-")
```

Comment:
Should we keep the hyphens instead of underscores in command line? This can be confusing when comparing with the config file.